File: backend/api/check-jobs.js
const admin = require("firebase-admin");
const serviceAccount = require("../../serviceAccountKey.json");

if (!admin.apps.length) {
  admin.initializeApp({
    credential: admin.credential.cert(serviceAccount)
  });
}
const db = admin.firestore();

async function checkJobs() {
  const snapshot = await db.collection("print_jobs").orderBy("createdAt", "desc").limit(3).get();
  snapshot.forEach(doc => {
    const data = doc.data();
    console.log(`Job ${doc.id} | Kiosk: ${data.kioskId} | Color: ${data.colorMode} | Status: ${data.status} | PrinterStatus: ${data.printerStatus || 'N/A'}`);
  });
  process.exit(0);
}

checkJobs().catch(console.error);

File: pi_scripts/check_jobs.py
import firebase_admin
from firebase_admin import credentials, firestore

try:
    cred = credentials.Certificate("/home/pi/mimo/serviceAccountKey.json")
    firebase_admin.initialize_app(cred)
    db = firestore.client()
    docs = db.collection("print_jobs").order_by("updatedAt", direction=firestore.Query.DESCENDING).limit(5).stream()
    for d in docs:
        data = d.to_dict()
        print(f"[{d.id}] Status: {data.get('status')} | Kiosk: {data.get('kioskId')} | Code: {data.get('printCode')} | Type: {data.get('colorMode')} | File: {data.get('fileName')}")
except Exception as e:
    print(f"Error: {e}")
